code/modules/power/cell.dm

```diff
@@ -148,6 +148,30 @@
 				if(prob(25))
 					corrupt()
 
+/obj/item/stock_parts/cell/attack_self(mob/user)
+	var/mob/living/carbon/human/H = user
+	if(istype(H))
+		var/datum/species/ethereal/eth_species = H.dna?.species
+		if(istype(eth_species))
+			if(charge < 25)
+				to_chat(H, "<span class='warning'>The [src] doesn't have enough power!</span>")
+				return
+			var/obj/item/organ/stomach/ethereal/stomach = H.getorganslot(ORGAN_SLOT_STOMACH)
+			if(stomach.crystal_charge > 96)
+				to_chat(H, "<span class='warning'>Your charge is full!</span>")
+				return
+			to_chat(H, "<span class='notice'>You clumsily channel power through the [src] and into your body, wasting some in the process.</span>")

```

Comment:
maybe add a way to channel charge from you into the cell?

Reply:
for some reason the idea of that sounds bad to me, like somewhere down the line people will bitch about the balance aspect of it, also im lazy so like maybe idk